src/interface/ContextInterface.tsx

```diff
@@ -237,3 +252,80 @@ export async function retrieveContextData(): Promise<boolean> {
     )
   );
 }
+
+function checkForObsoleteDiagrams() {
+  const diagramsInCache = Object.keys(CacheSearchVocabularies).flatMap(
+    (vocab) => CacheSearchVocabularies[vocab].diagrams
+  );
+  const diagramsWithVocabularies = Object.keys(CacheSearchVocabularies)
+    .filter(
+      (vocab) =>
+        vocab in WorkspaceVocabularies && !WorkspaceVocabularies[vocab].readOnly
+    )
+    .flatMap((vocab) => CacheSearchVocabularies[vocab].diagrams);
+  // Diagrams that
+  // ( are in cache
+  // *but* are not associated with write-enabled vocabularies present in the workspace )
+  const diff = _.difference(diagramsInCache, diagramsWithVocabularies);
+  const workspaceDiagrams = Object.values(Diagrams).map((diag) => diag.iri);
+  // *and* are in the workspace
+  // are to be deleted
+  const diagrams = _.intersection(diff, workspaceDiagrams);
```

Comment:
I am not sure I understand the logic here. And I am not sure how the cache works (or why there is one in the first place). But what we need is that if you have a blank workspace, add a vocabulary with a diagram, open it in OntoGrapher (that's where the cache gets populated I suppose), and then remove the vocabulary in Mission Control -> the vocabulary should disappear from OntoGrapher, including its diagram, *without* any further alert / notice.

For the next milestone we should probably discuss removing the cache altogether and reconstructing everything from just source data on the application load. As I said I don't have a good understanding of what it actually does, but it feels like something we should perhaps remove.

Reply:
The cache, in simple terms, is just a copy of "canon" SSP into graphDB. Without a doubt, we could be totally fine just querying the Virtuoso store, but the primary reason for the cache is (I assume) to take advantage of graphDB's connectors for searching.

Reply:
Update: I found the [document describing the cache architecture](https://docs.google.com/document/d/1C9IP8J4ZZ6BKqKZEerqM-Uj5z87Lfldh1v5Eec8JP5I/edit#) with information you might find interesting:

> Vzhledem k plánované [architektuře výrobní linky](https://docs.google.com/document/d/1nEiNCdyFkxlxcvujfqMKaiXQwBWKP9cqubPGYF1cJo4/edit) (VL), kdy slovníky v pracovním prostoru jsou automaticky editovatelné a všechny ostatní slovníky z SSP musí být k dispozici v read-only režimu a vzhledem k technickým omezením (TermIt není v tuto chvíli schopen pracovat s Virtuoso serverem, na kterém SSP běží) bylo rozhodnuto o vytvoření cache SSP, která by byla určena výhradně pro nástroje výrobní linky.

Reply:
@bindeali Thanks, I though it was yet another cache, in that case this if totally fine!